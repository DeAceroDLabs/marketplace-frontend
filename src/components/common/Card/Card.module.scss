@use "styles/colors";
@use "styles/typography";

.card {
  border-radius: 8px;
  margin: 17px 10px;
  border: 1px solid #e6e6e6;

  .cardImage {
    border-radius: 8px 8px 0px 0px;
    width: 100%;
    border-bottom: 1px solid #e6e6e6;
  }

  .cardInfo {
    font-size: typography.$body-m-size;
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 16px 20px;
    border-radius: 0px 0px 8px 8px;
    height: 54px;
  }
}

.small {
  width: 90%;
  max-width: 418px;
  min-width: 350px;
  box-shadow: 0px 16px 16px rgba(153, 153, 153, 0.15);
  .cardImage {
    height: 176.25px;
  }
  .cardInfo {
    height: 45.32px;
  }
}

.medium {
  width: 307px;
  box-shadow: 0px 16px 16px rgba(153, 153, 153, 0.15);

  .cardImage {
    height: 290px;
  }

  .cardInfo {
    height: 54px;
  }
}

.large {
  width: 417px;

  .cardImage {
    height: 432px;
  }

  .cardInfo {
    height: 32px;
  }

  .cardName {
    width: 50%;
  }

  .secondaryInfo {
    color: colors.$accent;
    font-weight: typography.$header-s-weight;
    width: 50%;
    text-align: right;
  }
}
