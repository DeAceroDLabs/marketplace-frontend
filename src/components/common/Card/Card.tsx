import styles from "./Card.module.scss";

interface CardProps {
  title: string;
  size?: "small" | "medium" | "large";
  loading?: "loading-true" | "loading-false";
  secondaryInfo?: string;
  imgSrc: string;
}

const Card: React.FC<CardProps> = ({
  title,
  size = "small",
  loading = "loading-false",
  imgSrc,
  secondaryInfo,
}) => {
  return (
    <div>
      <div className={`${styles.card} ${styles[size]} ${styles[loading]}`}>
        {(loading === "loading-false") && (
          <img className={styles.cardImage} src={imgSrc} alt="card" />
        )}
        {loading === "loading-true" && <div className={styles.cardImage} />}
        <div className={styles.cardInfo}>
          <div className={styles.cardName}> {title}</div>
          <div className={styles.secondaryInfo}> {secondaryInfo}</div>
        </div>
      </div>
    </div>
  );
};

export default Card;