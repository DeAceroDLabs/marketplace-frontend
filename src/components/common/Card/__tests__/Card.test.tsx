import renderer from "react-test-renderer";
import Card from "../Card";

const CardMockProps = {
  title: "Mock Title",
  imgSrc: "./mock/image/route",
};
it("renders Card with no issue", () => {
  const component = renderer.create(<Card {...CardMockProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
