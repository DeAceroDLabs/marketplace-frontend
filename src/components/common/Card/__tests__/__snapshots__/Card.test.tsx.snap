// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card with no issue 1`] = `
<div>
  <div
    className="card small "
  >
    <img
      alt="card"
      className="cardImage"
      src="./mock/image/route"
    />
    <div
      className="cardInfo"
    >
      <div
        className="cardName"
      >
         
        Mock Title
      </div>
      <div
        className="secondaryInfo"
      >
         
      </div>
    </div>
  </div>
</div>
`;
